vllm/engine/async_llm_engine.py

```diff
@@ -363,11 +363,18 @@ async def step_async(
                 self.cached_scheduler_outputs[
                     virtual_engine] = SchedulerOutputState()
 
+            current_step: Optional[int] = self._current_step(
+                seq_group_metadata_list)
+            # minus 1 because we updated the step count in finish_step() above.
+            step_num: Optional[
+                int] = current_step - 1 if current_step else None
```

Comment:
Can we simply pass `current_step` (i.e., `append_output(..., current_step=current_step)`) and let `append_output` deal with the minus 1 logic?

Reply:
No. We also pass step_idx_in_multi_step in `async_process_outputs` and `_final_process_outputs` in `multi_step_model_runner.py` - there, the "step_idx_from_multi_step" comes from the index of the output in `model_input.cached_outputs` . append_output wouldn't be able to differentiate between these calls to do the minus 1 correctly.

However, I realize that step_idx_in_multi_step is really used to check if the output belongs to the first step.  I can remove this argument and pass in `is_first_step_outputs` instead. What do you think ?

Reply:
Removed current_step and added `is_first_step_outputs` in its place.